diff --git a/src/shared/cvmix_kinds_and_types.F90 b/src/shared/cvmix_kinds_and_types.F90
index a8682db..10f3410 100644
--- a/src/shared/cvmix_kinds_and_types.F90
+++ b/src/shared/cvmix_kinds_and_types.F90
@@ -36,7 +36,7 @@ module cvmix_kinds_and_types
 
   ! Kind Types:
   ! The cvmix package uses double precision for floating point computations.
-  integer, parameter, public :: cvmix_r8       = selected_real_kind(15, 307), &
+  integer, parameter, public :: cvmix_r8       = selected_real_kind(6, 37), &
                                 cvmix_log_kind = kind(.true.),                &
                                 cvmix_strlen   = 256
 
diff --git a/src/shared/cvmix_math.F90 b/src/shared/cvmix_math.F90
index 64d6a30..0bc2d5a 100644
--- a/src/shared/cvmix_math.F90
+++ b/src/shared/cvmix_math.F90
@@ -180,15 +180,20 @@
 
   end subroutine cvmix_math_poly_interp
 
-  function cvmix_math_cubic_root_find(coeffs, x0)
+  function cvmix_math_cubic_root_find(coeffs_R8, x0_R8)
 
-    real(cvmix_r8), dimension(4), intent(in) :: coeffs
-    real(cvmix_r8),               intent(in) :: x0
+    real(cvmix_r8), dimension(4), intent(in) :: coeffs_R8
+    real(cvmix_r8),               intent(in) :: x0_R8
 
     real(cvmix_r8) :: cvmix_math_cubic_root_find
     real(cvmix_r8) :: fun_val, root, slope
     integer :: it_cnt
 
+    real(8), dimension(4) :: coeffs
+    real(8)              :: x0
+
+    coeffs = real(coeffs_R8, kind=8)
+    x0 = real(x0_R8, kind=8)
     root = x0
     fun_val = coeffs(4)*(root**3)+coeffs(3)*(root**2)+coeffs(2)*root+coeffs(1)
     do it_cnt = 1, CVMIX_MATH_MAX_NEWTON_ITERS
